nir/ir.py

```diff
@@ -120,6 +128,16 @@ class Delay(NIRNode):
     delay: np.ndarray  # Delay
 
 
+@dataclass
+class Flatten(NIRNode):
+    """Flatten node.
+
+    This node flattens the input tensor to a 1D tensor.
```

Comment:
Do we *always* want to flatten to a 1d tensor? Shouldn't this be specified somewhat like the PyTorch `Flatten` where you can select the dimensions to flatten?

For instance, something like `Flatten(start_dim, end_dim)`.

Reply:
For me the current version seems not really consistent or it needs more documentation. In the docstring, it says that the Flatten should flatten to a 1D Tensor. But then there is a parameter n_dims, "Number of dimensions to flatten". Is this supposed to select which dimensions should be flattened? Then the docstring is wrong. Or I am misunderstanding the parameter.
But I also like the proposal from @Jegp .